### PR TITLE
DUPLO-25186: Add logic to set IsAnyHostAllowed

### PR DESCRIPTION
**Pull request by ddiaz-duplo**
## Describe Changes

- Added logic to CronJob Update to set `isAnyHostAllowed`. We need to explicitly set this because of the choices made in the API design. Fixing the underlying problem would require making changes to our UI, Backend, and Terraform and deploying hotfixes to customers. 

## Link to Issues 

[DUPLO-25186](https://app.clickup.com/t/8655600/DUPLO-25186)

## PR Review Checklist  

- [x] Thoroughly reviewed on local machine.
- [x] Have you added any tests
- [x] Make sure to note changes in Changelog

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-25186 Duploctl overrides the nodeselector flag on UI](https://app.clickup.com/t/8655600/DUPLO-25186)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,178 +7,185 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.2.36] - 2024-09-24
```
Comment (kferrone) [inline CHANGELOG.md:10]: remove this line, you added unreleased features. The release process will inject this line when it actually bumps to 0.2.36

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,178 +7,185 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Added
+
+- Added logic to CronJob Update to set `isAnyHostAllowed`
```
Comment (kferrone) [inline CHANGELOG.md:14]: Isn't this a `fix` versus a new feature?

File: CHANGELOG.md
Comment (kferrone) [inline CHANGELOG.md:1]: Something is very wrong here. There are way too many changes in this file. Can you revert and make sure only the note about what you did is the only change to this file.

File: src/duplo_resource/cronjob.py [hunk 3]
```diff
@@ -8,38 +8,57 @@ class DuploCronJob(DuploTenantResourceV3):
...
-  
+
   @Command()
-  def update_schedule(self, 
-                      name: args.NAME, 
+  def update_schedule(self,
+                      name: args.NAME,
                       cronschedule: args.CRONSCHEDULE) -> dict:
     """Update the schedule of a cronjob.
-
```
Comment (kferrone) [inline src/duplo_resource/cronjob.py:35]: I'm thinking your editor did something funky here and makes this pr look way bigger than it is.

**Review by kferrone — CHANGES_REQUESTED** [review]

**Comment by kferrone** [comment]
This PR will not pass until you sign your commits. You will need to set this up and amend your previous commits with the signature. 

Here is how to do this: 
https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

I like to use 1passwords ssh feature to sign my commits, but not required as long as you sign them. Here is the 1pass docs on that: 
https://developer.1password.com/docs/ssh/git-commit-signing/

File: CHANGELOG.md
Comment (ddiaz-duplo) [reply CHANGELOG.md:1]: Its best to review with `Hide Whitespace` checked. I was resolving these warnings:
image
image

File: CHANGELOG.md (hunk 2 above)
Comment (ddiaz-duplo) [reply CHANGELOG.md]: fixed

**Review by kferrone — APPROVED** [review]

**Pull request merged**